src/content/developers/docs/data-structures/ssz/index.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Simple Serialize
+description: Explanation of Ethereum's SSZ format.
+lang: en
+sidebar: true
+sidebarDepth: 2
+---
+
+Simple serialize (SSZ) is the serialization method used on the Beacon Chain. It replaces the RLP serialization used on the execution layer everywhere across the consensus layer except the peer discovery protocol. SSZ is specifically designed to be deterministic and compatible with Merkleization.
+
+## How does SSZ work? {#how-does-ssz-work}
+
+### Serialization
+
+Ultimately the goal of SSZ serialization is to represent objects of arbitrary complexity as strings of bytes. Each bytestring has a fixed length (32 bytes) called a chunk. These chunks directly become leaves in the Merkle tree representing the object.
+
+This is a very simple process for "basic types". The element is simply converted to hexadecimal bytes and then right-padded until its length is equal to 32 bytes (little-endian representation). Basic types include:
+
+- unsigned integers
+- Booleans
+
+For more complex "composite" types the serialization is not quite so straightforward. This is because the composite type contains multiple elements that miught have different types or different sizes, or both. Vectors, for example, contain elements of uniform type but varying size. Where these objects all have fixed lengths (i.e. the size of the elements is always going to be constant irrespective of their actual values) the serialization is simply a conversion of each element in the composite type in order into little-endian bytestrings. These bytestrings are joined together and padded to the nearest multiple of 32-bytes. The serialized object has the bytelist representation of the fixed-length elements in the same order as they appear in the deserialized object.
+
+For types with variable lengths, the actual data is replecaed by an "offset" value in that element's position in the serialized object. The actual data is then added to a heap at the end of the serialized object. The offset value is the index for the start of the actual data in the heap, acting like a pointer to the relevant bytes.
+
+The example below illustrates how the offsetting works for an container with both fixed and variable length elements:
+
+```Rust
+
+    struct Dummy {
+
+        number: u64,
+        number2: u64,
+        vector: Vec<u8>
+        number3: u64
+    }
+
+    dummy = Dummy{
```

Comment:
Not familiar with rust. Is this formatting correct?

Reply:
the syntax is ok, but I missed a comma and had a name inconsistency between `number` and `number1` - now fixed.